### DISCUSSION TITLE
Drop-Shadow class breaks child absolute positioning

### DISCUSSION THREAD

**Discussion by DavidWhit** [tailwindlabs/tailwindcss#9029] [Help] [answered]
**What version of Tailwind CSS are you using?**
v3.1.7

**What build tool (or framework if it abstracts the build tool) are you using?**
"postcss": "^8.4.14", "node": "16.9.0", svelte kit

**What version of Node.js are you using?**
 "node": "16.9.0"

**What browser are you using?**
Chrome latest 

**What operating system are you using?**
Windows 10

```

<div class="drop-shadow-sm">
    <div class=" grid grid-cols-3 gap-1 place-items-center">
    <h1 class="text-xl">A</h1>
    <h1 class="text-xl">B</h1>
    <h1 class="text-xl">C</h1>
    <div class="fixed bottom-0 left-0 ...">
          </div>
          <label for="my-modal" class="btn modal-button h-9 btn-sm w-24 bg-red-500 hover:bg-gray-700  hover:text-white text-white">
            <i class="fa-duotone fa-trash mr-1"></i>
            Delete
            </label> 
            <input type="checkbox" id="my-modal" class="modal-toggle" />
            <div class="modal">
            <div class="modal-box">
                <h3 class="font-bold text-lg">Are you sure?</h3>
                <div class="modal-action">
                <label for="my-modal" class="btn btn-sm w-24 bg-green-500 text-white hover:bg-gray-700 hover:text-white" on:click={()=> console.log("asdfasdf")}>
                    <i class="fa-duotone fa-square-check mr-1"></i>
                    Yes</label>
                <label for="my-modal" class="btn btn-sm w-24 bg-red-500 text-white hover:bg-gray-700 hover:text-white">
                    <i class="fa-duotone fa-square-x mr-1"></i>
                    No</label>
                </div>
            </div>
            </div>
    </div>
</div>

```
**Describe your issue**
When adding the drop-shadow to an element it breaks child absolute/fixed positioning.  

Describe the problem you're seeing, any important steps to reproduce and what behavior you expect instead.
The modal appears inside the div but should be absolute visible outside the parent. To get the expected result remove the drop-shadow from the top div.

**Comment by wongjn** [comment]
I just created a [Tailwind play](https://play.tailwindcss.com/P0tyw8RKII) with your example but I couldn't replicate your issue – was there some more CSS in your project that you didn't include? Perhaps if you produced your own Tailwind Play and then we can look at diagnosing the problem?

**Comment by DavidWhit** [comment]
Sorry, I could have made that even simpler to see without having the daisy ui modal. 
Here is a another [Tailwind Play](https://play.tailwindcss.com/YBSCxTLrEG) that is more concise.

**Comment by wongjn** [comment, marked as answer]
The `drop-shadow` utilities use the `drop-shadow()` function in the `filter` CSS property. In the [specification for the `filter` property](https://www.w3.org/TR/filter-effects-1/#FilterProperty), there's this line:

> A value other than none for the [filter](https://www.w3.org/TR/filter-effects-1/#propdef-filter) property results in the creation of a [containing block](https://www.w3.org/TR/css-display-3/#containing-block) for absolute and fixed positioned descendants unless the element it applies to is a document root element in the current [browsing context](https://html.spec.whatwg.org/multipage/browsers.html#browsing-context).

So it looks like the behavior is by-design in CSS. I think the only work around is to not use `drop-shadow` or any other `filter` property and try to achieve your goal another way, such as using the `shadow` utility instead.

**Reply by DavidWhit** [reply]
Thanks, that solution is what I did. I just had a hard time trying to resolve it until I found the cause; oddly enough, setting a height to either child also fixes it.
